Common/stHashIterator.hpp

```diff
@@ -0,0 +1,170 @@
+#ifndef STHASH__ITERATOR_H
+#define STHASH__ITERATOR_H 1
+
+#include <string>
+#include <limits>
+#include "nthash.hpp"
+
+
+/**
+ * Iterate over hash values for k-mers in a
+ * given DNA sequence.
+ *
+ * This implementation uses ntHash
+ * function to efficiently calculate
+ * hash values for successive k-mers.
+ */
+
+class stHashIterator
+{
+
+public:
+   
+    static std::vector<std::vector<unsigned> > parseSeed(const std::vector<std::string> &seedString) {
+        std::vector<std::vector<unsigned> > seedSet;
+        for (unsigned i=0; i< seedString.size(); i++) {
+            std::vector<unsigned> sSeed;
+            for (unsigned j=0; j < seedString[i].size(); j++) {
+                if(seedString[i][j]!='1') sSeed.push_back(j);
+            }
+            seedSet.push_back(sSeed);
+        }
+        return seedSet;
+    }
+    
+    /**
+     * Default constructor. Creates an iterator pointing to
+     * the end of the iterator range.
+    */
+    stHashIterator():
+        m_hVec(NULL),
+        m_hStn(NULL),
+        m_pos(std::numeric_limits<std::size_t>::max())
+    {}
+
+    /**
+     * Constructor.
+     * @param seq address of DNA sequence to be hashed
+     * @param seed address of spaced seed
+     * @param k k-mer size
+     * @param h number of hashes
```

Comment:
Should specify that `h` is the number of seeds, and `h2` is the number of hashes per seed.